packages/react-component-library/src/components/DatePicker/useInput.ts

```diff
@@ -26,7 +25,7 @@ export function useInput(
 ) {
   const handleKeyDown = useCallback(
     (event: React.KeyboardEvent<HTMLInputElement>) => {
-      if (isRange || event.keyCode !== RETURN) {
+      if (isRange || event.key !== 'Enter') {
```

Comment:
`user-event` no longer sets `keyCode` as it's deprecated, so I've switched to `key` (which works on older browsers, `code` doesn't work in IE11).